v2/internal/reconcilers/arm/arm_client.go

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package arm
+
+import (
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/Azure/azure-service-operator/v2/internal/genericarmclient"
+)
+
+type armClient struct {
```

Comment:
minor: A comment on this type might help explain what exactly it is: It holds the client as well as the secret used to create it and the credential from which that secret was retrieved (or something like that?)